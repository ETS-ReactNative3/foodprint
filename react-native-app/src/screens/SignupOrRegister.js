import React, { useState } from 'react';
import { View, SafeAreaView } from 'react-native';
import { Text, Button } from 'react-native-elements';
import AntDesign from 'react-native-vector-icons/AntDesign';
import { AuthContext } from '../store/Auth';


export default SignupOrRegister = ({ navigation }) => {
  const { signInWithGoogle, signInAnonymously } = React.useContext(AuthContext);
  const [isPressed, setIsPressed] = useState(false);

  const handleSignInWithGoogle = async () => {
    setIsPressed(true);
    await signInWithGoogle();
    setIsPressed(false);
  };

  const handleSignInAnonymously = async () => {
    setIsPressed(true);
    await signInAnonymously();
    setIsPressed(false);
  };

  return (
    <SafeAreaView style={{ justifyContent: 'center', alignItems: 'center', height: '100%' }}>
      <View style={{ position: 'absolute', top: 32, alignItems: 'center', marginTop: 32 }}>
        <Text h1>
          FoodPrint
        </Text>
      </View>
      <View style={{ width: '80%' }}>
        <View>
          <Button
            disabled={isPressed}
            iconContainerStyle={{ marginRight: 24 }}
            icon={
              <AntDesign
                style={{ paddingRight: 5 }}
                name="google"
                size={28}
                color="black"
              />
            }
            containerStyle={{ marginVertical: 8 }}
            buttonStyle={{ backgroundColor: 'white' }}
            titleStyle={{ fontSize: 24, color: 'black' }}
            title="Join using Google"
            onPress={handleSignInWithGoogle}
          />
          <Button
            disabled={isPressed}
            containerStyle={{ marginVertical: 8 }}
            buttonStyle={{ backgroundColor: 'green' }}
            titleStyle={{ fontSize: 24 }}
            title="Join using email"
            onPress={() => navigation.navigate('Signup')}
          />
          <View style={{ marginVertical: 12 }}>
            <Text style={{ fontSize: 18, textAlign: 'center', color: 'grey' }}>Existing user?</Text>
            <Button
              title="LOGIN"
              disabled={isPressed}
              titleStyle={{ color: 'green', fontSize: 24 }}
              onPress={() => navigation.navigate('Login')}
              type="clear"
            />
            <Button
              disabled={isPressed}
              title="skip"
              titleStyle={{ color: 'grey', fontSize: 24 }}
              onPress={handleSignInAnonymously}
              type="clear"
            />
          </View>
        </View>
      </View>
    </SafeAreaView >
  );
}
  ;
