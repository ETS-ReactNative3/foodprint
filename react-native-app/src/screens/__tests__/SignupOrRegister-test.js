import React from 'react';

import renderer from 'react-test-renderer';
import SignupOrRegister from '../SignupOrRegister';
import { fireEvent, render, wait } from '@testing-library/react-native';
import { AuthContext } from '../../store/Auth';

jest.mock('@apollo/react-hooks');

// Check current component against its snapshot
test('SignupOrRegister matches previous snapshot', () => {
  const authentication = {
    signInWithGoogle: () => { },
    signInAnonymously: () => { },
  };

  const tree = renderer.create(<AuthContext.Provider value={authentication}>
    <SignupOrRegister />
  </AuthContext.Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('GoogleSignIn button calls SignInWithGoogle from context', async () => {
  const signInWithGoogle = jest.fn()
  const authentication = {
    signInWithGoogle
  };

  const utils = renderSignUpOrRegister(authentication)

  const googleButton = utils.getByText('Join using Google')
  fireEvent.press(googleButton)
  await wait(() => expect(utils.queryByTestId('googleButton')));
})

test('Skip button calls anonymous sign in', async () => {
  const signInAnonymously = jest.fn()
  const authentication = {
    signInAnonymously
  };

  const utils = renderSignUpOrRegister(authentication)

  const skipButton = utils.getByText('skip')
  fireEvent.press(skipButton)
  await wait(() => expect(utils.queryByTestId('skipButton')));
  expect(signInAnonymously.mock.calls.length).toBe(1);
})

// HELPERS
// Re-use this function to render Signup, as it uses useContext. This 'mocks' that out.
function renderSignUpOrRegister(authentication) {
  return render(
    <AuthContext.Provider value={authentication}>
      <SignupOrRegister />
    </AuthContext.Provider>
  );
}

