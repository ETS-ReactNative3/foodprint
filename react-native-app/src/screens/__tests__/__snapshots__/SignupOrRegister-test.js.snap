// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupOrRegister matches previous snapshot 1`] = `
<SafeAreaView
  style={
    Object {
      "alignItems": "center",
      "height": "100%",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 32,
        "position": "absolute",
        "top": 32,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 50,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      FoodPrint
    </Text>
  </View>
  <View
    style={
      Object {
        "width": "80%",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "borderRadius": 3,
            "marginVertical": 8,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.3}
          disabled={false}
          onPress={[Function]}
          testID="googleButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 28,
                  },
                  Object {
                    "paddingRight": 5,
                  },
                  Object {
                    "fontFamily": "anticon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              î¦µ
            </Text>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 24,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Join using Google
            </Text>
          </View>
        </TouchableOpacity>
      </View>
      <View
        style={
          Object {
            "borderRadius": 3,
            "marginVertical": 8,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.3}
          disabled={false}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "green",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 24,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Join using email
            </Text>
          </View>
        </TouchableOpacity>
      </View>
      <View
        style={
          Object {
            "marginVertical": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 18,
              "textAlign": "center",
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          Existing user?
        </Text>
        <View
          style={
            Object {
              "borderRadius": 3,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.3}
            disabled={false}
            onPress={[Function]}
            testID="skipButton"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "green",
                    "fontSize": 24,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                LOGIN
              </Text>
            </View>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 32,
        "position": "absolute",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.3}
        disabled={false}
        onPress={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "grey",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            skip
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </View>
</SafeAreaView>
`;
